### PR TITLE
Ignore my other PR that was a mistake i promise 

### PR DESCRIPTION
**Pull request by UnknownAMD**
my finger slipped and selected master on the last pr

anyways this pr is better cuz it's not just stupid stuff, I finished the JFusion interface

### CODE REVIEW

File: Plugins/CTFAK.Decompiler/FTDecompile.cs [hunk 1]
```diff
@@ -200,16 +200,19 @@ public void Execute(IFileReader reader)
                 var key = game.frameitems.Keys.ToArray()[i];
                 var item = game.frameitems[key];
                 var newItem = TranslateObject(mfa,game,item);
+                //Object Section
                 if (newItem.Loader == null)
                 {
-                    throw new NotImplementedException("Unsupported Object: "+newItem.ObjectType);
+                    throw new NotImplementedException("Fucked Object: "+newItem.ObjectType);
```
Comment (REDxEYE) [inline Plugins/CTFAK.Decompiler/FTDecompile.cs:206]: Please revert back to normal error message

File: Tools/JFusion/Program.cs [hunk 2]
```diff
@@ -30,44 +30,101 @@ public static void Main(string[] args)
...
+                    string jpath = string.Empty;
+                    Console.Write("Json path: ");
+                    jpath = Console.ReadLine().Trim('"');
+                    if (!File.Exists(jpath))
+                    {
+                        Console.ForegroundColor = ConsoleColor.Red;
+                        Console.WriteLine("ERROR: Project not found");
+                        goto JSONPATH;
+                    }
+                    MFAOUT:
```
Comment (REDxEYE) [inline Tools/JFusion/Program.cs:94]: Can we avoid using goto statements? Wrap into while loop or into separate function

File: Tools/JFusion/Program.cs [hunk 3]
```diff
@@ -30,44 +30,101 @@ public static void Main(string[] args)
...
+                    break;
+                
+
                 case ConsoleKey.D3:
                     var testMfa = new MFAData();
                     testMfa.Read(new ByteReader("test.mfa",FileMode.Open));
                     testMfa.Write(new ByteWriter("test_out.mfa",FileMode.Create));
                     break;
-                default:
+                    default:
```
Comment (REDxEYE) [inline Tools/JFusion/Program.cs:121]: Formatting error

**Review by REDxEYE — CHANGES_REQUESTED** [review]

File: Plugins/CTFAK.Decompiler/FTDecompile.cs (hunk 1 above)
Comment (UnknownAMD) [reply Plugins/CTFAK.Decompiler/FTDecompile.cs:206]: ok

File: Tools/JFusion/Program.cs (hunk 2 above)
Comment (UnknownAMD) [reply Tools/JFusion/Program.cs:94]: idk i saw kostya using them so i kept using them

File: Tools/JFusion/Program.cs (hunk 3 above)
Comment (UnknownAMD) [reply Tools/JFusion/Program.cs:121]: my bad

File: Tools/JFusion/Program.cs (hunk 2 above)
Comment (UnknownAMD) [reply Tools/JFusion/Program.cs]: > Can we avoid using goto statements? Wrap into while loop or into separate function

Done.

**Review by UnknownAMD — COMMENTED** [review]
very epic indeed

File: Tools/JFusion/Program.cs [hunk 4]
```diff
@@ -1,88 +1,152 @@
...
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine(ex);
+            while (true)
+            {
+                if (Console.ReadKey().Key == ConsoleKey.Enter) break;
+            }
+        }
+        goto SELECT_MODE;
+
```
Comment (REDxEYE) [inline Tools/JFusion/Program.cs:142]: In this case, you can get rid of while loops completely, code loops itself by it's on

**Comment by AITYunivers** [comment]
Too many commits for 1 PR alone. PR isn't clear on what it does. Closing.

**Pull request closed without merging**